app/src/main/java/openfoodfacts/github/scrachx/openfood/views/BaseActivity.java

```diff
@@ -36,4 +41,24 @@ public void setContentView(@LayoutRes int layoutResID) {
     public ActivityComponent getActivityComponent() {
         return activityComponent;
     }
+
+    @Override
+    public void onBackPressed() {
+        if (doubleBackToExitPressedOnce) {
+            super.onBackPressed();
+            return;
+        }
+
+        this.doubleBackToExitPressedOnce = true;
+        Toast.makeText(this, "Please press back again to exit", Toast.LENGTH_SHORT).show();
+
+        new Handler().postDelayed(new Runnable() {
+
+            @Override
+            public void run() {
+                doubleBackToExitPressedOnce=false;
+            }
+        }, 1000);
```

Comment:
@AbsolutelySaurabh I guess the timer of the postDelayed `run` method should be same as the duration of Toast Message and as the duration of LENGTH_SHORT is 2000ms so I'd suggest doing same in `run` method too. :)

Reply:
Ya, I've changed this too in the new PR #777 .